std/selector/multiplexer_test.go

```diff
@@ -0,0 +1,224 @@
+
+	// Ignoring the circuit's output:
+	assert.ProverSucceeded(&ignoredOutputMuxCircuit{}, &ignoredOutputMuxCircuit{SEL: 0, I0: 0, I1: 1, I2: 2})
+
+	assert.ProverSucceeded(&ignoredOutputMuxCircuit{}, &ignoredOutputMuxCircuit{SEL: 2, I0: 0, I1: 1, I2: 2})
+
+	// Failures
+	assert.ProverFailed(&ignoredOutputMuxCircuit{}, &ignoredOutputMuxCircuit{SEL: 3, I0: 0, I1: 1, I2: 2})
+
+	assert.ProverFailed(&ignoredOutputMuxCircuit{}, &ignoredOutputMuxCircuit{SEL: -1, I0: 0, I1: 1, I2: 2})
+}
+
+// Map tests:
+type mapCircuit struct {
+	SEL            frontend.Variable
+	K0, K1, K2, K3 frontend.Variable
+	V0, V1, V2, V3 frontend.Variable
+	OUT            frontend.Variable
+}
+
+func (c *mapCircuit) Define(api frontend.API) error {
+
+	out := selector.Map(api, c.SEL,
+		[]frontend.Variable{c.K0, c.K1, c.K2, c.K3},
+		[]frontend.Variable{c.V0, c.V1, c.V2, c.V3})
+
+	api.AssertIsEqual(out, c.OUT)
+
+	return nil
+}
+
+type ignoredOutputMapCircuit struct {
+	SEL    frontend.Variable
+	K0, K1 frontend.Variable
+	V0, V1 frontend.Variable
+}
+
+func (c *ignoredOutputMapCircuit) Define(api frontend.API) error {
+
+	_ = selector.Map(api, c.SEL,
+		[]frontend.Variable{c.K0, c.K1},
+		[]frontend.Variable{c.V0, c.V1})
+
+	return nil
+}
+
+func TestMap(t *testing.T) {
+	assert := test.NewAssert(t)
+	assert.ProverSucceeded(&mapCircuit{},
+		&mapCircuit{
+			SEL: 100,
+			K0:  100, K1: 111, K2: 222, K3: 333,
+			V0: 0, V1: 1, V2: 2, V3: 3,
+			OUT: 0,
+		})
+
+	assert.ProverSucceeded(&mapCircuit{},
+		&mapCircuit{
+			SEL: 222,
+			K0:  100, K1: 111, K2: 222, K3: 333,
+			V0: 0, V1: 1, V2: 2, V3: 3,
+			OUT: 2,
+		})
+
+	assert.ProverSucceeded(&mapCircuit{},
+		&mapCircuit{
+			SEL: 333,
+			K0:  100, K1: 111, K2: 222, K3: 333,
+			V0: 0, V1: 1, V2: 2, V3: 3,
+			OUT: 3,
+		})
+
+	// Duplicated key, success:
+	assert.ProverSucceeded(&mapCircuit{},
+		&mapCircuit{
+			SEL: 333,
+			K0:  222, K1: 222, K2: 222, K3: 333,
+			V0: 0, V1: 1, V2: 2, V3: 3,
+			OUT: 3,
+		})
+
+	// Duplicated key, UNDEFINED behavior: (with our hint implementation it fails)
+	assert.ProverFailed(&mapCircuit{},
+		&mapCircuit{
+			SEL: 333,
+			K0:  100, K1: 111, K2: 333, K3: 333,
+			V0: 0, V1: 1, V2: 2, V3: 3,
+			OUT: 3,
+		})
+
+	assert.ProverFailed(&mapCircuit{},
+		&mapCircuit{
+			SEL: 77,
+			K0:  100, K1: 111, K2: 222, K3: 333,
+			V0: 0, V1: 1, V2: 2, V3: 3,
+			OUT: 3,
+		})
+
+	assert.ProverFailed(&mapCircuit{},
+		&mapCircuit{
+			SEL: 111,
+			K0:  100, K1: 111, K2: 222, K3: 333,
+			V0: 0, V1: 1, V2: 2, V3: 3,
+			OUT: 2,
+		})
+
+	// Ignoring the circuit's output:
+	assert.ProverSucceeded(&ignoredOutputMapCircuit{},
+		&ignoredOutputMapCircuit{SEL: 5,
+			K0: 5, K1: 7,
+			V0: 10, V1: 11,
+		})
+
+	assert.ProverFailed(&ignoredOutputMapCircuit{},
+		&ignoredOutputMapCircuit{SEL: 5,
+			K0: 5, K1: 5,
+			V0: 10, V1: 11,
+		})
+
+	assert.ProverFailed(&ignoredOutputMapCircuit{},
+		&ignoredOutputMapCircuit{SEL: 6,
+			K0: 5, K1: 7,
+			V0: 10, V1: 11,
+		})
+
+}
+
+func Example() {
```

Comment:
I recommend removing this example. First it creates additional files in the working tree and I think that the examples can be more specific for mux and map functions. I created these new doc-examples which I think are more suitable.

Reply:
I agree. We should definitely remove it. I just wrote it to check the number of constraints and then I kept it. However it can be problematic, with every edit in the code it breaks! 

The examples that you've written are great.